src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -353,7 +353,9 @@ class EditorObjectSelectionWindow final : public Window
         switch (widgetIndex)
         {
             case WIDX_CLOSE:
-                WindowClose(*this);
+                if (!CheckObjectSelection() && !(gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER))
```

Comment:
Can swap these conditions, so CheckObjectSelection does not need to be executed in the track designs manager case:  
```suggestion
                if (!(gScreenFlags & SCREEN_FLAGS_TRACK_MANAGER) && !CheckObjectSelection())
```

Reply:
Good catch.